File: benchmarks/read-file/read.erl
-module(read).
-export([main/0]).

read(File) ->
  case file:read_line(File) of
    {ok, Data} -> [Data | read(File)];
    eof        -> []
  end.

main() ->
  {ok, File} = file:open("../data/big.txt", [read]),
  {Time, _} = timer:tc(fun read/1, [File]),
  io:format("~p\n", [Time]),
  file:close(File).

File: benchmarks/read-file/read.py
import aiofiles
import asyncio
import time

wc = 0


async def run():
    global wc
    async with aiofiles.open('../data/big.txt') as f:
        async for line in f:
            wc += 1


def main():
    loop = asyncio.get_event_loop()
    start_time = time.perf_counter()
    loop.run_until_complete(run())
    print(((time.perf_counter() - start_time) * 1000 * 1000, wc))
    loop.close()


if __name__ == '__main__':
    main()
